freebsd-stubs.h

```diff
@@ -0,0 +1,14 @@
+#ifndef _FREEBSD_STUBS_H
+#define _FREEBSD_STUBS_H
+
+#define O_PATH 0
+#define O_DSYNC 0
+#define O_LARGEFILE 0
+#define O_NOATIME 0
+#define O_TMPFILE 0
```

Comment:
How about just

```diff
--- i/document-portal/document-portal-fuse.c
+++ w/document-portal/document-portal-fuse.c
@@ -286,8 +295,10 @@ open_flags_to_string (int flags)
     g_string_append (s, ",ASYNC");
   if (flags & O_FSYNC)
     g_string_append (s, ",FSYNC");
+#ifdef O_DSYNC
   if (flags & O_DSYNC)
     g_string_append (s, ",DSYNC");
+#endif
   if (flags & O_CREAT)
     g_string_append (s, ",CREAT");
   if (flags & O_TRUNC)
@@ -298,16 +309,22 @@ open_flags_to_string (int flags)
     g_string_append (s, ",CLOEXEC");
   if (flags & O_DIRECT)
     g_string_append (s, ",DIRECT");
+#ifdef O_LARGEFILE
   if (flags & O_LARGEFILE)
     g_string_append (s, ",LARGEFILE");
+#endif
+#ifdef O_NOATIME
   if (flags & O_NOATIME)
     g_string_append (s, ",NOATIME");
+#endif
   if (flags & O_NOCTTY)
     g_string_append (s, ",NOCTTY");
   if (flags & O_PATH)
     g_string_append (s, ",PATH");
+#ifdef O_TMPFILE
   if (flags & O_TMPFILE)
     g_string_append (s, ",TMPFILE");
+#endif

   return g_string_free (s, FALSE);
 }
```

and considering my other comments, we wouldn't need a `freebsd-stubs.h` at all